From: Gunnar Wrobel <p@rdus.de>
Subject: [PATCH] t/kronolith/HK/GW/AuthenticatedFreeBusy

Allow the use of authenticated Free/Busy with Kolab.

FIXME: Check if that has been commited/submitted upstream.

Signed-off-by: Gunnar Wrobel <p@rdus.de>

---
 config/conf.xml  |    6 ++++++
 lib/FreeBusy.php |    3 +++
 2 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/config/conf.xml b/config/conf.xml
index aa72858..0ed2387 100644
--- a/config/conf.xml
+++ b/config/conf.xml
@@ -49,6 +49,12 @@
   </configswitch>
  </configsection>
 
+ <configsection name="authenticated_freebusy">
+   <configheader>Use authenticated free/busy</configheader>
+   <configboolean name="enable" desc="Should we authenticate with the current user credentials agains the free/busy URL?"
+   required="false">true</configboolean>
+ </configsection>
+
  <configsection name="metadata">
   <configheader>Metadata Settings</configheader>
   <configboolean name="keywords" desc="Should keywords be loaded from
diff --git a/lib/FreeBusy.php b/lib/FreeBusy.php
index a0615dc..a8db0ee 100644
--- a/lib/FreeBusy.php
+++ b/lib/FreeBusy.php
@@ -187,6 +187,9 @@ class Kronolith_FreeBusy {
 
             require_once 'HTTP/Request.php';
             $http = new HTTP_Request($url, $options);
+            if (!empty($GLOBALS['conf']['authenticated_freebusy'])) {
+                $http->setBasicAuth(Auth::getAuth(), Auth::getCredential('password'));
+            }
             if (is_a($response = @$http->sendRequest(), 'PEAR_Error')) {
                 return PEAR::raiseError(sprintf(_("The free/busy url for %s cannot be retrieved."), $email));
             }
-- 
tg: (6938161..) t/kronolith/HK/GW/AuthenticatedFreeBusy (depends on: master)
-- 
TOPGIT patch commit log
=======================

commit b7ae3fffce45d1ec54a9477a0bc006e995d0abf0
Author: Gunnar Wrobel <p@rdus.de>
Date:   Sun Feb 1 18:19:43 2009 +0000

    Remove stuff from broken patching.

commit ebb26e5d1fbfae64d93da8fc83c82589a82812f2
Author: Gunnar Wrobel <p@rdus.de>
Date:   Sun Feb 1 18:18:21 2009 +0000

    Added patch kronolith/HK-GW-Authenticated_FB.patch from the mercurial release queue.
